toucan_connectors/toucan_connector.py

```diff
@@ -227,20 +225,27 @@ def retry_decorator(self):
         return RetryPolicy(**kwargs)
 
     @abstractmethod
-    def _retrieve_data(self, data_source: ToucanDataSource):
+    def _retrieve_data(self, data_source: ToucanDataSource, **kwargs):
         """Main method to retrieve a pandas dataframe"""
 
     @decorate_func_with_retry
     def get_df(
         self,
         data_source: ToucanDataSource,
         permissions: Optional[dict] = None,
+        **kwargs,
     ) -> pd.DataFrame:
         """
         Method to retrieve the data as a pandas dataframe
         filtered by permissions
         """
-        res = self._retrieve_data(data_source)
+        # This conditional prevents passing secrets in kwargs to connectors that can't use them
+        # inspect module's signature object can't be used here in Python 3.6 because it is not
+        # generated the same way as in 3.7 and 3.8; this can be eventually replaced
```

Comment:
Thanks for commenting on this!
Should we rather use the python version check in the `if` so it's more obvious?
If some coverage issue happen, it's OK to add a `pragma` comment to indicate that this branch will not be covered.